megamek/src/megamek/client/ui/swing/unitSelector/ASAdvancedSearchPanel.java

```diff
@@ -0,0 +1,709 @@
+    JLabel damageEBetween = new JLabel(BETWEEN);
+    MMComboBox<ASDamage> damageEFrom = new MMComboBox<>("Damage E From");
+    JLabel damageEAnd = new JLabel(AND);
+    MMComboBox<ASDamage> damageETo = new MMComboBox<>("Damage E To");
+
+    JCheckBox usePV = new JCheckBox("Point Value");
+    JLabel pvBetween = new JLabel(BETWEEN);
+    IntRangeTextField pvFrom = new IntRangeTextField();
+    JLabel pvAnd = new JLabel(AND);
+    IntRangeTextField pvTo = new IntRangeTextField();
+
+    JCheckBox useMV = new JCheckBox("Movement");
+    MMComboBox<String> mvMode = new MMComboBox<>("Movement Mode");
+    JLabel mvBetween = new JLabel(BETWEEN);
+    IntRangeTextField mvFrom = new IntRangeTextField();
+    JLabel mvAnd = new JLabel(AND);
+    IntRangeTextField mvTo = new IntRangeTextField();
+
+    JCheckBox useAbility1 = new JCheckBox("Special Ability");
+    MMComboBox<BattleForceSUA> ability1 = new MMComboBox<>("Ability1", BattleForceSUA.values());
+
+    JCheckBox useAbility2 = new JCheckBox("Special Ability");
+    MMComboBox<BattleForceSUA> ability2 = new MMComboBox<>("Ability2", BattleForceSUA.values());
+
+    public ASAdvancedSearchPanel() {
+        setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+        add(unitTypePanel());
+        add(new DottedSeparator());
+        add(simplePanel(useSize, size1, size2, size3, size4, size5));
+        add(new DottedSeparator());
+        add(simplePanel(usePV, pvBetween, pvFrom, pvAnd, pvTo));
+        add(new DottedSeparator());
+        add(simplePanel(useMV, mvMode, mvBetween, mvFrom, mvAnd, mvTo));
+        add(new DottedSeparator());
+        add(simplePanel(useTMM, tmm0, tmm1, tmm2, tmm3, tmm4, tmm5));
+        add(new DottedSeparator());
+        add(simplePanel(useArmor, armorBetween, armorFrom, armorAnd, armorTo));
+        add(new DottedSeparator());
+        add(simplePanel(useStructure, structureBetween, structureFrom, structureAnd, structureTo));
+        add(new DottedSeparator());
+        add(simplePanel(useThreshold, thresholdBetween, thresholdFrom, thresholdAnd, thresholdTo));
+        add(new DottedSeparator());
+        add(simplePanel(useDamageS, damageSBetween, damageSFrom, damageSAnd, damageSTo));
+        add(simplePanel(useDamageM, damageMBetween, damageMFrom, damageMAnd, damageMTo));
+        add(simplePanel(useDamageL, damageLBetween, damageLFrom, damageLAnd, damageLTo));
+        add(simplePanel(useDamageE, damageEBetween, damageEFrom, damageEAnd, damageETo));
+        add(new DottedSeparator());
+        add(simplePanel(useOV, ov0, ov1, ov2, ov3, ov4));
+        add(new DottedSeparator());
+        add(simplePanel(useAbility1, ability1));
+        add(simplePanel(useAbility2, ability2));
+        initializeCombos();
+        updateEnabled();
+    }
+
+    /** @return True when the given MechSummary matches the active search filters or true when no filters are active. */
+    public boolean matches(MechSummary mechSummary) {
```

Comment:
This method should really be using chained if -> else if -> else statements to prevent future misreads.